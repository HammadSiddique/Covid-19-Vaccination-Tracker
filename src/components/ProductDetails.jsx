import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { fetchProductDetail, removeSelectedProduct } from '../redux/products/products';

const ProductDetails = () => {
  const { productId } = useParams();
  const product = useSelector((state) => state.product);
  const {
    image, title, price, category, description,
  } = product;
  const dispatch = useDispatch();
  useEffect(() => {
    if (productId && productId !== '') dispatch(fetchProductDetail(productId));
    return () => {
      dispatch(removeSelectedProduct());
    };
  }, []);
  return (
    <div className="product-container">
      {Object.keys(product).length === 0 ? (
        <div className="loading-container"><div className="custom-loader" /></div>
      ) : (
        <div className="product-detail-container">
          <div className="details-img">
            <img className="item-image" src={image} alt={title} />
          </div>
          <div className="details-content">
            <h2 className="item-name">{title}</h2>
            <h3>
              <p className="item-price">
                $
                {price}
              </p>
            </h3>
            <h4 className="item-category">{category}</h4>
            <p className="item-description">{description}</p>
          </div>
        </div>
      )}
    </div>
  );
};

export default ProductDetails;
