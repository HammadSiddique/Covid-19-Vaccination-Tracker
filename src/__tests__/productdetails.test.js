import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import ProductDetails from '../components/ProductDetails';
import store from '../redux/configureStore';

it('renders properly', () => {
  const tree = renderer.create(
    <Router>
      <Provider store={store}>
        <ProductDetails />
      </Provider>
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
