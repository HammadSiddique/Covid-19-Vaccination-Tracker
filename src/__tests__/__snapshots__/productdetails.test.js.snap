// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="product-container"
>
  <div
    className="loading-container"
  >
    <div
      className="custom-loader"
    />
  </div>
</div>
`;
